rechunker/api.py

```diff
@@ -10,6 +13,84 @@
 from rechunker.algorithm import rechunking_plan
 
 
+class Rechunked(Delayed):
```

Comment:
Long-term, we might not want to subclass Dask Delayed, and instead make our own little light collection:

We don't necessarily want all the magic of `dask.delayed` creating new task graphs on `.<arbitrary_attribute>`. And if you do anything else to the Rechunked result, I think you get back a `Delayed` object, which won't have `execute`.

Reply:
Would we want this class documented in the docs?